clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -456,6 +456,20 @@ public static String formatAddress(String host, Integer port) {
         return sb.toString();
     }
 
+    public static <K, V> String join(Map<K, V> map,
```

Comment:
Hmm, interesting.  It looks like Utils.java already has both Utils#mkString(Collection<T> coll, String separator) and Utils#join(Collection<T> list, String seperator).  The mkString version returns a CharSequence instead of a String.  I'm guessing that the reason is increased efficiency-- the StringBuilder object itself implements CharSequence, whereas converting a StringBuilder into a String requires copying.  This could get a little tricky, so perhaps we should leave it alone for now, or at least open a separate JIRA for it?